master/internal/agent/agent.go

```diff
@@ -198,6 +208,11 @@ func (a *agent) containerStateChanged(ctx *actor.Context, sc aproto.ContainerSta
 }
 
 func (a *agent) summarize(ctx *actor.Context) model.AgentSummary {
+	agentState := ctx.Ask(
```

Comment:
nit: could just store these in the agent struct when we tell the slots/rm about these changes.

Reply:
it sounds odd to me to duplicate that state. having `disabled` state both in slots and agents sounds okay because both agents and slots can currently be disabled individually (although I am not sure if disabling slots has actual product value).
and here we'd need to maintain it just for convenience? I feel like that goes against the point of responsibility separation of actors.

Reply:
I think we'd maintain it because the agent is the ultimate owner of its state but yeah also for convenience; like, the agent is draining and it just tells the resource pool this so it won't schedule on it, but ultimately its the agent that's draining and knows its draining? in general, the ask's are just a little dicey, too, since they have potential to deadlock.

Reply:
also, that means this ask will happen N times for every call to `/agents`, which is kinda a lot of calls in some cases to go through serial execution, esp for an endpoint each webui calls every 5s? i mean, it probably would be really fast in general still, but eh.

Reply:
okay fine. I'll leave a TODO to make the entire thing better.